### ISSUE TITLE
expecting exact match even when num_memory_shards is 1

### ISSUE THREAD

**Issue by jhoguet** [typesense/typesense#315]
## Description

In implementing the work around from https://github.com/typesense/typesense/issues/304#issuecomment-872307238 to reduce our num_memory_shards from the default of 4 to 1, to honor `typo_tokens_threshold=1` across all memory shards, we surfaced another issue. 

We have a collection that includes high schools, and a high school called `GED`. 

When we search `ged` against our collection that has `num_memory_shards` at the default of 4, we find `GED`. 

But when we search out collection that has `num_memory_shards` at 1 we do not. 

## Steps to reproduce

We were able to reproduce by adding an exact match and a suffix match

```
import * as Typesense from 'typesense';

const config = require('../config-staging.json');

const typesense = new Typesense.Client({
    'nodes': [{ 'host': config.TYPESENSE_HOST, 'port': '443', 'protocol': 'https' }],
    'apiKey': config.TYPESENSE_ADMIN_KEY
});

(async () => {
    const schemaName = 'ged-bug-v2'
    await typesense.collections().create({
        "num_memory_shards": 1,
        "name": schemaName,
        "fields": [
            { "name": "InstitutionID", "type": "string" },
            { "name": "InstitutionName", "type": "string" },
            { "name": "PhysicalCity", "type": "string" },
            { "name": "PhysicalState", "type": "string" }
        ]
    });

    await typesense.collections(schemaName).documents().import([
        { InstitutionID: 'my-ged', InstitutionName: 'GED', PhysicalCity: '', PhysicalState: '' },
        {
            InstitutionID: '685034',
            InstitutionName: 'Yeshivah Gedolah High School',
            PhysicalCity: 'Oak Park',
            PhysicalState: 'MI',
            id: '22444'
        }
    ]);

    const result = await typesense.collections(schemaName).documents().search({
        'query_by': 'InstitutionName,PhysicalCity,PhysicalState',
        'q': 'ged',
        per_page: 10,
        limit_hits: 10,
        typo_tokens_threshold: 1
    })

    console.log(result.hits.map(h => h.document));

})();


```

## Expected Behavior

expect search `ged` to find `InstitutionName: 'GED' `

## Actual Behavior

It finds `InstitutionName: 'Yeshivah Gedolah High School',` and does not find `InstitutionName: 'GED' `

## Metadata

0.21.0

**Comment by kishorenc** [comment]
Thanks, I'm taking a look at this. Will get back to you soon.

**Comment by kishorenc** [comment]
@jhoguet Can you please try against this Docker build: `typesense/typesense:0.22.0.rc2`?

**Comment by jasonbosco** [comment]
Released in v0.22
